### PR TITLE
Supports for MongoDB

### PR DESCRIPTION
**Pull request by xieqihui**
Add supports to store index data in MongoDB.

### CODE REVIEW

File: nearpy/engine.py [hunk 1]
```diff
@@ -136,32 +140,38 @@ def neighbours(self, v):
...
         """ Collect candidates from all buckets from all hashes """
         candidates = []
         for lshash in self.lshashes:
             for bucket_key in lshash.hash_vector(v, querying=True):
-                bucket_content = self.storage.get_bucket(lshash.hash_name,
-                                                         bucket_key)
+                bucket_content = self.storage.get_bucket(
+                    lshash.hash_name,
+                    bucket_key,
+                    mongo_fetch_vector_filters
```
[the remark this answers is not in the document]
Comment (pixelogik) [reply nearpy/engine.py]: @xieqihui Do you have interest / time to look into the above?

[the remark this answers is not in the document]
Comment (xieqihui) [reply nearpy/engine.py]: Yes, I will look into your suggestion.

[the remark this answers is not in the document]
Comment (pixelogik) [reply nearpy/engine.py]: @xieqihui Great, thanks!

[the remark this answers is not in the document]
Comment (xieqihui) [reply nearpy/engine.py]: @pixelogik 
`mongo_fetch_vector_filters` is a dict that enables us to filter candidate on MongoDB, unlike other `vector_filters` which only conduct filtering after retrieving the results from the storage. This adds the capability to retrieve candidates from MongoDB not only based on hash, but also on the original features. To support filtering on DB, `store_vector()` in `MongoStorage` supports to store the key-value pairs of data if it is a dict. 

This added functionality is useful in one of my projects where I want to get the nearest neighbours satisfying some additional conditions, e.g., find similar houses whose price is less than 1 million. This is in fact another advantage of using MongoDB  comparing to Redis, besides the memory consumption.

If you see the value of adding such a functionality, could you consider to accept the difference in the signature of `get_bucket` and the change of signature in `neighbour()` in `Engine`? Anyway, the  `mongo_fetch_vector_filters` is defaulted to `None`, so adding it won't affect other storage type. 

Nevertheless. I have followed your suggestion to remove `mongo_fetch_vector_filters` completely in commit `37ef7ef`. This version works the same way as other storage types, i.e., only retrieve candidates based on hash. Please review it to see if it's worthy to keep it as this.

**Pull request merged**